src/components/input/select/index.tsx

```diff
@@ -242,7 +259,7 @@ export class SmoothlyInputSelect implements Input, Editable, Clearable, Componen
 	}
 	render(): VNode | VNode[] {
 		return (
-			<Host tabIndex={0} class={{ "has-value": this.selectedElement ? true : false }}>
+			<Host tabIndex={0} class={{ "has-value": this.selected.length > 0 ? true : false }}>
```

Comment:
`class={{ "has-value": this.selected.length }}` should be enough. If it isn't use `class={{ "has-value": this.selected.length != 0 }}`